### DISCUSSION TITLE
Make Linux Builds Auto-Update

### DISCUSSION THREAD

**Discussion by NekomusuMae** [ppy/osu#14653] [Ideas] [answered]
Can we make the Linux build of the game auto-update so Linux users don't have to manually redownload and reinstall the game everytime the game updates?
https://docs.appimage.org/packaging-guide/optional/updates.html#making-appimages-self-updateable

**Comment by peppy** [comment, marked as answer]
Linux users are a small percentage of our user base so this is well on the bucket list currently.

That said, you are more than welcome to chip in and add support!

**Comment by Diyou** [comment]
A solution could be be to ship a prebuilt [appimageupdatetool](https://github.com/AppImage/AppImageUpdate) AppImage or selfmade binary(to reduce bloat) inside the osu appimage and run it like
```bash
/usr/bin/appimageupdatetool-x86_64.AppImage --overwrite /path/to/osu.AppImage
```
With
```csharp
Process.start
```
[ProcessStartInfo.RedirectStandardOutput](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.redirectstandardoutput?view=net-6.0#system-diagnostics-processstartinfo-redirectstandardoutput) may be helpful to parse the progress.

**Comment by Schweeeeeeeeeeeeeeee** [comment]
How about some auto updates now?
It's been about 3/4 of a year how's the current situation, is it still not far up on your bucket list, or will it be added in 2 or three months 3?

**Reply by Schweeeeeeeeeeeeeeee** [reply]
Didn't know what zsync was, so I never knew they already implemented it. oopsie
